### PR TITLE
Load process.json for exec and add detach

### PR DESCRIPTION
**Pull request by crosbymichael**
This change adds loading a process.json for exec via a flag so that you can exec processes using both methods.  

It also adds detach and the pid-file to exec so that runc can exit after it spawns the processes.

### CODE REVIEW

File: start.go [hunk 1]
```diff
@@ -209,3 +188,39 @@ func destroy(container libcontainer.Container) {
...
+	if detach {
+		if err := dupStdio(process, rootuid); err != nil {
+			return nil, err
+		}
+		return nil, nil
+	}
+	return createStdioPipes(process, rootuid)
+}
+
+func createPidile(path string, process *libcontainer.Process) error {
```
Comment (mrunalp) [inline start.go]: Typo: createPidFile

**Comment by hqhq** [comment]
The first commit called `setupIO`, but it's added in the second commit, that'll break bisect, can you reroll these commit?

File: start.go (hunk 1 above)
Comment (crosbymichael) [inline start.go]: fxied

File: exec.go [hunk 2]
```diff
@@ -53,72 +65,94 @@ var execCommand = cli.Command{
...
 	container, err := getContainer(context)
 	if err != nil {
 		return -1, err
 	}
-
-	bundle := container.Config().Rootfs
-	if err := os.Chdir(path.Dir(bundle)); err != nil {
+	var (
+		detach = context.Bool("detach")
+		bundle = container.Config().Rootfs
```
Comment (hqhq) [inline exec.go]: I would prefer s/bundle/rootfs :-)

**Comment by hqhq** [comment]
A minor tip, otherwise LGTM

**Comment by crosbymichael** [comment]
@hqhq fixed.  thanks!

File: exec.go [hunk 3]
```diff
@@ -53,72 +65,94 @@ var execCommand = cli.Command{
...
+	tty, err := setupIO(process, rootuid, context.String("console"), p.Terminal, detach)
 	if err != nil {
 		return -1, err
 	}
+	if err := container.Start(process); err != nil {
+		return -1, err
+	}
+	if pidFile := context.String("pid-file"); pidFile != "" {
+		if err := createPidFile(pidFile, process); err != nil {
+			return -1, err
```
Comment (LK4D4) [inline exec.go]: Shouldn't we kill a process in case of failure? Because it seems for me like broken flow.

File: start.go [hunk 4]
```diff
@@ -209,3 +188,39 @@ func destroy(container libcontainer.Container) {
...
+}
+
+func createPidFile(path string, process *libcontainer.Process) error {
+	pid, err := process.Pid()
+	if err != nil {
+		return err
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		logrus.WithField("pid", pid).Error("create pid file")
```
Comment (LK4D4) [inline start.go]: I don't think it should be just log. If user wants pid-file, they probably want to fail on its creation.

**Comment by crosbymichael** [comment]
@LK4D4 fixed

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**